### PR TITLE
feat(pair_stable): added initial amount

### PR DESCRIPTION
**Pull request by P-Yevhenii**

### CODE REVIEW

File: contracts/pair_stable/src/contract.rs [hunk 1]
```diff
@@ -526,15 +545,16 @@ pub fn provide_liquidity(
 
         let after_fee_d = compute_d(amp, &new_balances, config.greatest_precision)?;
 
-        Decimal256::with_precision(total_share, config.greatest_precision)?
+        let share = Decimal256::with_precision(total_share, config.greatest_precision)?
```
Comment (ShadoySV) [inline contracts/pair_stable/src/contract.rs:548]: Maybe just use **checked_multiply_ratio** on total_share?

File: contracts/pair_stable/src/contract.rs [hunk 2]
```diff
@@ -496,8 +496,27 @@ pub fn provide_liquidity(
...
+            deps.querier,
+            &config,
+            &env.contract.address,
+            &env.contract.address,
+            MINIMUM_LIQUIDITY_AMOUNT,
+            false,
+        )?);
+
+        // share cannot become zero after minimum liquidity subtraction
+        if share.is_zero() {
```
Comment (epanchee) [inline contracts/pair_stable/src/contract.rs:515]: This check is repeated in else-branch. I suggest to move it outside the if-else clause (499-557).

**Review by epanchee — CHANGES_REQUESTED** [review]

File: contracts/pair_stable/src/contract.rs (hunk 2 above)
Comment (ShadoySV) [reply contracts/pair_stable/src/contract.rs:515]: That's due to different error messages

Comment (epanchee) [reply contracts/pair_stable/src/contract.rs:515]: Ah, ok

**Review by epanchee — APPROVED** [review]

**Pull request merged**